include/networkit/graph/GraphTools.hpp

```diff
@@ -418,6 +418,27 @@ void sortEdgesByWeight(Graph &G, bool decreasing = false);
  */
 std::vector<node> topologicalSort(const Graph &G);
 
+/**
+ * Given a  grah G, randomizes the weights in the graph. The weights are distributed in
+ * the range [0, 1] by default, unless a distribution is provided. This occurs in-place
+ * if the graph is already weighted, but creates a weighted copy before randomizing weights
+ * if it wasn't.
+ * @param   G           Directed input graph
+ * @param   distr       Random distribution
+ */
+template <class Distribution = std::uniform_real_distribution<edgeweight>>
+void randomizeWeights(Graph &G, Distribution distr = std::uniform_real_distribution<edgeweight>{
+                                    0, std::nexttoward(1.0, 2.0)}) {
+    if (!G.isWeighted())
+        G = toWeighted(G);
+    std::vector<std::mt19937> gens(omp_get_max_threads());
+    for (std::mt19937 gen : gens) {
+        gen.seed(Aux::Random::getSeed());
+    }
```

Comment:
Shouldn't you use a reference here?
```suggestion
    for (auto &gen : gens) {
        gen.seed(Aux::Random::getSeed());
    }
```